the-collab-lab/tcl-46-smart-shopping-list
# Untrack list in home and enable list deletion in multilist manager

## Description
- Created UI for deleting lists in Home
- Moved isValidToken check to utils
- light renaming and headings across pages based on current layout planning
- Updated untracking functionality and moved list management functions into ListSwitcher component. Now when user clicks "untrack" button next to a list, the list should disappear from both UI and the locally stored user token list. Current list should go to the next available list token or show NoToken UI.

## Related Issue

Closes #71 and #72 

## Acceptance Criteria

- [x] Untrack list works for each list in the ListSwitcher component.
- [x] Delete list option is added to ListSwitcher for each tracked list, on the Home page.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓   | :bug: Bug fix              |
|    | :sparkles: New feature     |
|  ✓   | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before (deletion UI changes)

in List page
![issue_delete](https://user-images.githubusercontent.com/102257735/186473238-d3463898-a2f8-4ead-bd12-9f5d86c87557.png)

in Home page
![issue_delete_home](https://user-images.githubusercontent.com/102257735/186473404-2fd4a12b-904a-4445-82c7-663a56e46b02.png)

### After
![issue_addDelete_home](https://user-images.githubusercontent.com/102257735/186552164-723b3a75-6119-44ac-81b1-24e665e1f79b.png)

![issue_delete_after](https://user-images.githubusercontent.com/102257735/186551970-b633884a-a270-4572-8a18-93ff5bddcf7c.png)

## Testing Steps / QA Criteria
On branch `untrackListInHome`, add lists, open local storage, and try untracking and deleting using the buttons in the Manage Lists section on the Home page.

When on List page, delete list option is no longer present.
